samples/SimpleDataSource/HelloWorldCustomDataProcessor.cs

```diff
@@ -0,0 +1,159 @@
+﻿using Microsoft.Performance.SDK;
+using Microsoft.Performance.SDK.Extensibility;
+using Microsoft.Performance.SDK.Processing;
+using SampleCustomDataSource.Tables;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SampleCustomDataSource
+{
+    //
+    // This is a sample Custom Data Processor that processes simple text files.
+    //
+    // Custom Data Processors are created in Custom Data Source and are used to actually process the files.
+    // An instance of the Processor is created for each file you open whereas only one instance of the Custom Data Source is ever created.
+    //
+    // Data processor is responsible for instantiating the proper tables based on what the user has decided to enable.
+    //
+
+    //
+    // Derive the CustomDataProcessorBase abstract class.
+    //
+
+    public sealed class HelloWorldCustomDataProcessor
+        : CustomDataProcessorBase
+    {
+        private readonly string[] filePaths;
+        private IReadOnlyDictionary<string, IReadOnlyList<Tuple<Timestamp, string>>> fileContent;
+        private DataSourceInfo dataSourceInfo;
```

Comment:
I would explain in a comment what this thing is. Since we're doing that, might as well annotate each field here with a comment :)